lib/igb.c

```diff
@@ -615,6 +622,15 @@ igb_reset(struct adapter *adapter)
 			rxdctl |= IGB_RX_HTHRESH << 8;
 			rxdctl |= IGB_RX_WTHRESH << 16;
 			E1000_WRITE_REG(hw, E1000_RXDCTL(i), rxdctl);
+
+			rxdctlpollcnt = 0;
```

Comment:
Any particular reason why you don't like to use for loop here?

Reply:
On the other hand do  {} while(); would look nice since you could skip break there.

Reply:
I've converted it into _do{} while()_ as suggested, looks a bit better :-)